test/e2e/network/service.go

```diff
@@ -1450,6 +1450,10 @@ var _ = SIGDescribe("ESIPP [Slow]", func() {
 		// requires cloud load-balancer support - this feature currently supported only on GCE/GKE
 		framework.SkipUnlessProviderIs("gce", "gke")
 
+		// Skipping this test for too large clusters due to issue #52495.
+		// TODO(MrHohn): Get rid of this when gce-side load-balancer improvements are done.
+		framework.SkipUnlessNodeCountIsAtMost(framework.GCPMaxInstancesInInstanceGroup)
```

Comment:
Thanks for the PR!

I am hesitate about which exact tests should we skip --- not all of ESIPP tests will create LB (like `should work for type=NodePort` does not). Also, should we skip other non-ESIPP tests that also create LB (like `should only allow access from service loadbalancer source ranges` and `should be able to change the type and ports of a service`)?

Reply:
All the tests that seem to be failing (https://k8s-testgrid.appspot.com/sig-scalability-gce#gce-scale-correctness) are also the ones in this suite. So I think this is fine.

Reply:
Oh.. But I think you're right about the type=NodePort one.
Wrt the others on LB, let's keep them for now as they're passing.

Reply:
So do you suggest adding this check for each of the e2e's except that one?

Reply:
I think it is fine :) It may be too messy to separate them.